lib/InstModel/GenericInstModel/GenericInstModel.cpp

```diff
@@ -60,8 +62,15 @@ std::vector<Token *> GenericInstModel::extractToken(llvm::StoreInst *Inst) {
   TokenVec.push_back(
       this->getTokenWrapper()->getToken(Inst->getPointerOperand()));
   llvm::Value *ValOp = Inst->getValueOperand();
-  if (!llvm::isa<llvm::ConstantInt>(ValOp))
-    TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));
+  if (!llvm::isa<llvm::ConstantInt>(ValOp)){
+    if(llvm::isa<llvm::ConstantPointerNull>(ValOp)){
+      std::string s("NULL");
+      TokenVec.push_back(this->getTokenWrapper()->getToken(s,nullptr));
+    }
+    else{
+      TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));
+    }
+  }
```

Comment:
Can this be written as:
```cpp
  if (!llvm::isa<llvm::ConstantInt>(ValOp)) return TokenVec;      

  if (llvm::isa<llvm::ConstantPointerNull>(ValOp))
    TokenVec.push_back(this->getTokenWrapper()->getToken("NULL",nullptr));
  else
    TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));
```

Reply:
I think you are trying to say
`if (llvm::isa<llvm::ConstantInt>(ValOp)) return TokenVec;      

  if (llvm::isa<llvm::ConstantPointerNull>(ValOp))
    TokenVec.push_back(this->getTokenWrapper()->getToken("NULL",nullptr));
  else
    TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));`
And if it is then I have done this. Can I do this more general way like instead of using llvm::ConstantInt that can use only for integer constant we can use llvm::Constant which is more general way to say constant because constant can be float or double also.

Reply:
I think you are trying to say
```C
if (llvm::isallvm::ConstantInt(ValOp)) return TokenVec;

if (llvm::isallvm::ConstantPointerNull(ValOp))
TokenVec.push_back(this->getTokenWrapper()->getToken("NULL",nullptr));
else
TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));
```
And if it is then I have done this. Can I do this more general way like instead of using llvm::ConstantInt that can use only for integer constant we can use llvm::Constant which is more general way to say constant because constant can be float or double also.

Reply:
I think you are trying to say
`if (llvm::isa<llvm::ConstantInt>(ValOp)) return TokenVec;

if (llvm::isa<llvm::ConstantPointerNull>(ValOp))
TokenVec.push_back(this->getTokenWrapper()->getToken("NULL",nullptr));
else
TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));`
And if it is then I have done this. Can I do this more general way like instead of using ```llvm::ConstantInt``` that can use only for integer constant we can use ```llvm::Constant``` which is more general way to say constant because constant can be float or double also.

Reply:
I think you are trying to say
```C
if (llvm::isa<llvm::ConstantInt>(ValOp)) return TokenVec;

if (llvm::isa<llvm::ConstantPointerNull>(ValOp))
TokenVec.push_back(this->getTokenWrapper()->getToken("NULL",nullptr));
else
TokenVec.push_back(this->getTokenWrapper()->getToken(ValOp));
```
And if it is then I have done this. Can I do this more general way like instead of using **llvm::ConstantInt** that can use only for integer constant we can use **llvm::Constant** which is more general way to say constant because constant can be float or double also.

Reply:
Yes, please :)

Reply:
I just realized that there can be other places where ConstantInt is used and it will fail with floats and doubles (thanks for pointing that out). Can you please create a github issue for that?